src/autopas/tuning/tuningStrategy/decisionTreeTuning/DecisionTreeTuning.cpp

```diff
@@ -0,0 +1,168 @@
+/**
+ * @file DecisionTreeTuning.cpp
+ * @author Abdulkadir Pazar
+ * @date 20.06.24
+ */
+
+#include "DecisionTreeTuning.h"
+
+#include <Python.h>
+
+#include <iostream>
+#include <sstream>
+
+namespace autopas {
+
+// Constructor
+DecisionTreeTuning::DecisionTreeTuning(const std::set<Configuration> &searchSpace, const std::string &modelFileName)
+    : _configurations(searchSpace), _modelFileName(modelFileName), _pFunc(nullptr) {
+  // Initialize Python Interpreter
+  Py_Initialize();
+  loadScript();
+}
+
+// Destructor
+DecisionTreeTuning::~DecisionTreeTuning() {
+  // Finalize Python Interpreter
+  if (_pFunc) {
+    Py_DECREF(_pFunc);  // Clean up the reference to _pFunc
+  }
+  Py_Finalize();
+}
+
+void DecisionTreeTuning::loadScript() {
+  // Add the current directory to the Python path
+  PyObject *sysPath = PySys_GetObject((char *)"path");
+  PyObject *cwd = PyUnicode_FromString(".");
+  PyList_Append(sysPath, cwd);
+  Py_DECREF(cwd);
+
+  // Load the fixed Python script predict.py
+  PyObject *pName = PyUnicode_DecodeFSDefault("predict");
+  if (!pName) {
+    throw std::runtime_error("Failed to convert script name to Python string.");
+  }
+
+  PyObject *pModule = PyImport_Import(pName);
```

Comment:
Use a smart pointer: https://en.cppreference.com/book/intro/smart_pointers. I am not super familiar as to what impact this might have on the memory leak issue.

Reply:
Since the Python objects themselves are managed by Python's own memory management system, primarily through reference counting, it is a bit unconventional to use custom deleters with smart pointers to handle PyObject reference counting.

Reply:
Would prefer implementing own deleter required for smart pointer.

Reply:
Migrated to pybind.